runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -928,9 +893,49 @@ public synchronized CorfuRuntime connect() {
                     .setTimeoutInMinutesForLoading((int) parameters.fastLoaderTimeout.toMinutes());
             fastLoader.loadMaps();
         }
+
+        gcThread.execute(this::runRuntimeGC);
+
         return this;
     }
 
+    /**
+     * Over time garbage is created on the global log and clients clients sync
+     * parts of the log therefore clients can accumulate garbage overtime. This
+     * method runs a garbage collection process on the different client views,
+     * which discards data before the trim mark (i.e. the point that demarcates
+     * the beginning of the active log).
+     */
+    private void runRuntimeGC() {
+        for(;;){
+            try {
+                Sleep.MINUTES.sleepRecoverably(parameters.getRuntimeGCPeriod().toMinutes());
+                long currTrimMark = getAddressSpaceView().getTrimMark().getSequence();
+                if (trimMark == currTrimMark) {
+                    log.debug("runRuntimeGC: TrimMark({}) hasn't changed, skipping.");
+                    continue;
+                }
+
+                log.debug("runRuntimeGC: starting gc cycle, removing {} to {}", trimMark,
+                        currTrimMark);
+                long startTs = System.currentTimeMillis();
+                getObjectsView().gc(currTrimMark);
+                getAddressSpaceView().gc(currTrimMark);
+                long endTs = System.currentTimeMillis();
+                trimMark = currTrimMark;
```

Comment:
You don't need to have `trimMark` as a field in this class, you can pass it as a parameter in your method